src/chain/elders_info.rs

```diff
@@ -29,7 +29,7 @@ use std::{
 #[derive(Default, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
 pub struct EldersInfo {
     /// The complete list of the section's elders' IDs.
-    members: BTreeSet<P2pNode>,
+    members: BTreeMap<PublicId, P2pNode>,
```

Comment:
Looking at the follow up PR, Probably better to do that change here and use XorName as the key directly, this way you won't have to jump through many hoops in follow up PR to find P2pNode for a XorName.

Reply:
Makes sense, I'll have a go at that